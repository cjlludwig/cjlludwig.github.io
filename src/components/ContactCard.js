import headshot from '../static/images/headshot.jpg';
import '../static/components/ContactCard.css';

function Card() {
  return (
    <div className='card-wrapper'>
      <div className="card">
        <div className="top">
          <h2 className="name">Connor J Ludwig</h2>
          <img className="circle-img" src={headshot} alt="avatar_img" />
        </div>
        <div className="bottom">
          <p className="info">
            <b>Cell: </b>
            (XXX)-XXX-XXXX
          </p>
          <p className="info">
            <b>Email: </b>
            <a href = "mailto: cjlludwig@gmail.com">cjlludwig@gmail.com</a>
          </p>
          <p className="info">
            <b>Linkedin: </b>
            <a href="www.linkedin.com/in/connor-ludwig/">www.linkedin.com/in/connor-ludwig/</a>
          </p>
        </div>
      </div>
    </div>
  );
}

export default Card;