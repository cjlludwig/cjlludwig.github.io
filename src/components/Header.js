import { useState, useEffect, useCallback, useRef } from 'react';
import { Nav, Navbar } from 'react-bootstrap';
import personalLogo from '../static/images/logo_transparent512.png';

import "../static/styles/components/Header.css";

function Header() {
  const [ navBarClass, setNavBarClass ] = useState("header-navbar-scrolling");
  const [ pixelsScrolled, setPixelsScrolled ] = useState(window.scrollY);
  const [ isSticky, setSticky ] = useState(false);
  const [height, setHeight] = useState(0);
  const navbarRef = useRef(null);
  const viewportHeight = window.innerHeight;

  const handleNavigation = useCallback(e => {
    const window = e.currentTarget;
    
    if (window.scrollY <= viewportHeight) {
      console.log("hit");
      setNavBarClass("header-navbar-scrolling");
      setSticky(false);
    }

    if (window.scrollY >= viewportHeight) {
      console.log("hit");
      setNavBarClass("header-navbar");
      setSticky(true);

    }
    setPixelsScrolled(window.scrollY);
  }, [ pixelsScrolled, viewportHeight ]);

  useEffect(() => {
    setPixelsScrolled(window.scrollY);
    window.addEventListener("scroll", handleNavigation);

    return () => {
      window.removeEventListener("scroll", handleNavigation);
    };
  }, [handleNavigation]);

  useEffect(() => {
    setHeight(navbarRef.current.clientHeight)
  }, [])

  return ( 
    <>
      <Navbar  
        ref={navbarRef}
        className={navBarClass}
      > 
        <Navbar.Brand href="/">
          <img className="header-logo" src={personalLogo} alt="Personal logo"/>
        </Navbar.Brand>
        <Navbar.Toggle aria-controls="basic-navbar-nav" />
        <Navbar.Collapse id="basic-navbar-nav">
          <Nav className="mr-auto" >
            <Nav.Link key= '1' eventKey='1' href="#home" style={{ color: 'white' }}>Home</Nav.Link>
            <Nav.Link key= '2' href="#aboutMe" style={{ color: 'white' }}>About Me</Nav.Link>
            <Nav.Link key= '3' href="#resume" style={{ color: 'white' }}>Resume</Nav.Link>
            <Nav.Link key= '4' href="#links" style={{ color: 'white' }}>Links</Nav.Link>
            <Nav.Link key= '5' href="#contactCard" style={{ color: 'white' }}>Contact Card</Nav.Link>
          </Nav>
        </Navbar.Collapse>
      </Navbar>
      {isSticky &&
        <div style={{ height: `${height}px`}}/>
      }
    </>
  );
}

export default Header;