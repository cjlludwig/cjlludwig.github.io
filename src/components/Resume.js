import { Document, Page, pdfjs } from 'react-pdf'
import { useState } from 'react';

import '../static/components/Resume.css'
import 'react-pdf/dist/esm/Page/AnnotationLayer.css';
import 'react-pdf/dist/esm/Page/TextLayer.css';

import resumePdf from '../static/images/cjludwig_resume.pdf'

pdfjs.GlobalWorkerOptions.workerSrc = `//cdnjs.cloudflare.com/ajax/libs/pdf.js/${pdfjs.version}/pdf.worker.js`;

function Resume() {
  const [numPages, setNumPages] = useState(null);
  const [pageNumber, setPageNumber] = useState(1);

  function onDocumentLoadSuccess({ numPages }) {
    setNumPages(numPages);
    setPageNumber(1);
  }

  function changePage(offset) {
    setPageNumber(prevPageNumber => prevPageNumber + offset);
  }

  function previousPage() {
    changePage(-1);
  }

  function nextPage() {
    changePage(1);
  }

  return (
    <>
      <Document file={resumePdf} onLoadSuccess={onDocumentLoadSuccess}>
        <Page 
          key={1}
          className={pageNumber === 1 ? "prevPage" : ""}
          pageNumber={1} 
        />
        <Page
          key={2}
          pageNumber={2}
          className={pageNumber === 2 ? "prevPage" : ""}
        />
      </Document>
      <div className='pdf-buttons'>
        <button
          type="button"
          disabled={pageNumber <= 1}
          onClick={previousPage}
          >
          {"<"}
        </button>
        <span>
          {pageNumber || (numPages ? 1 : '--')} of {numPages || '--'}
        </span>
        <button
          type="button"
          disabled={pageNumber >= numPages}
          onClick={nextPage}
        >
          {">"}
        </button>
      </div>
    </>
  );
}

export default Resume;