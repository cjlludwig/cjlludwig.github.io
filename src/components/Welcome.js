import Typewriter from 'typewriter-effect';
import '../static/styles/components/Welcome.css';

function Welcome() {
  return (
    <div className='typewriter-wrapper'>
      <Typewriter
        onInit={(typewriter) => {
          typewriter.typeString('Hello!')
            .pauseFor(300)
            .deleteAll()
            .typeString("\nMy name's Connor.<br>")
            .pauseFor(300)
            .typeString("\nWelcome to my site!<br>")
            .pauseFor(2000)
            .typeString("\nScroll on for more...")
            .start();
        }}
      />
    </div>
  );
}

export default Welcome;

