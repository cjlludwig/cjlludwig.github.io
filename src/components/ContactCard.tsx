import headshot from '../static/images/cjl_headshot.png';
import '../static/styles/components/ContactCard.css';

function Card() {
  return (
    <div className='contact-card-wrapper'>
      <div className="contact-card">
        <div className="top">
          <h2 className="name">Connor J Ludwig</h2>
          <img className="circle-img" src={headshot} alt="avatar_img" />
        </div>
        <div className="bottom">
          <p className="info">
            <b>Cell: </b>
            &emsp;&emsp;(712)-830-6202
          </p>
            <b>Email: </b>
            &emsp;&ensp;<a href = "mailto: cjlludwig@gmail.com">cjlludwig@gmail.com</a>
          <p className="info">
          </p>
          <p className="info">
            <b>Github: </b>
            &emsp;<a href="https://github.com/cjlludwig">https://github.com/cjlludwig</a>
          </p>
          <p className="info">
            <b>Linkedin: </b>
            &nbsp;<a href="www.linkedin.com/in/connor-ludwig/">www.linkedin.com/in/connor-ludwig/</a>
          </p>
        </div>
      </div>
    </div>
  );
}

export default Card;