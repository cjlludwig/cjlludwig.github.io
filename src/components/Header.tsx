import { useState, useEffect, useCallback, useRef } from 'react';
import { Nav, Navbar } from 'react-bootstrap';
import personalLogo from '../static/images/logo_transparent512.png';
import 'bootstrap/dist/css/bootstrap.min.css';


import "../static/styles/components/Header.css";

function Header() {
  const [navBarClass, setNavBarClass] = useState("header-navbar-scrolling");
  const [pixelsScrolled, setPixelsScrolled] = useState(window.scrollY);
  const [isSticky, setSticky] = useState(false);
  const [height, setHeight] = useState(0);
  const navbarRef = useRef<HTMLDivElement>(null);
  const viewportHeight = window.innerHeight;
  const isMobile = window.innerWidth <= 960; // Common mobile cutoff

  const handleNavigation = useCallback((e: { currentTarget: any }) => {
    const window = e.currentTarget;
    if (isMobile) {
      if (pixelsScrolled <= height) {
        setNavBarClass("header-navbar-scrolling-mobile");
        setSticky(false);
      }
      if (pixelsScrolled >= height) {
        setNavBarClass("header-navbar-mobile");
        setSticky(true);
      }
    } else {
      if (pixelsScrolled <= viewportHeight) {
        setNavBarClass("header-navbar-scrolling");
        setSticky(false);
      }
      if (pixelsScrolled >= viewportHeight) {
        setNavBarClass("header-navbar");
        setSticky(true);
      }
    }
    setPixelsScrolled(window.scrollY);
  }, [pixelsScrolled, viewportHeight, height, isMobile]);

  useEffect(() => {
    setPixelsScrolled(window.scrollY);
    window.addEventListener("scroll", handleNavigation);

    return () => {
      window.removeEventListener("scroll", handleNavigation);
    };
  }, [handleNavigation]);

  useEffect(() => {
    if (navbarRef.current) {
      setHeight(navbarRef.current.clientHeight)
    }
  }, [])

  return (
    <>
      <Navbar
        ref={navbarRef}
        className={navBarClass}
      >
        <Navbar.Brand href="/">
          <img className="header-logo" src={personalLogo} alt="Personal logo" />
        </Navbar.Brand>
        <Navbar.Toggle aria-controls="basic-navbar-nav" />
        <Nav className="mr-auto" >
          <Nav.Link key='nav-1' eventKey='1' href="#home" style={{ color: 'white' }}>Home</Nav.Link>
          <Nav.Link key='nav-2' href="#aboutMe" style={{ color: 'white' }}>About Me</Nav.Link>
          <Nav.Link key='nav-3' href="#resume" style={{ color: 'white' }}>Resume</Nav.Link>
          <Nav.Link key='nav-4' href="#listening_to" style={{ color: 'white' }}>Listening To</Nav.Link>
          <Nav.Link key='nav-6' href="#contactCard" style={{ color: 'white' }}>Contact Card</Nav.Link>
          <Nav.Link key='nav-5' href="#links" style={{ color: 'white' }}>Links</Nav.Link>
        </Nav>
      </Navbar>
      {isSticky &&
        <div style={{ height: `${height}px` }} />
      }
    </>
  );
}

export default Header;