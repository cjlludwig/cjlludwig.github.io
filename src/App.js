import logo from './static/images/logo.svg';
import nodeLogo from './static/images/Node.js_logo.svg';
import gitLogo from './static/images/GitHub_Logo.png';
import dockerLogo from './static/images/docker.svg';
import vsCodeLogo from './static/images/vscode.svg';
import personalLogo from './static/images/logo_transparent512.png';
import awsPic from './static/images/ludwig_aws.JPG';

import ContactCard from './components/ContactCard';
import Resume from './components/Resume';

import './static/pages/App.css';
// import FadeInSection from './components/FadeInSection';


function App() {

  return (
    <div className="App">
      <div className='img-1'>
        <h1 className='img-text-1'>Hello!</h1>
      </div>
      <div className='section-1'>
        <div className='opening'>
          <span className='opening-span'>
            <div className='img-holder'>
              <img src={awsPic} className="aws-pic" alt="AWS Pic" />
            </div>
            <div>
              <div>
                <b>My name's Connor.</b>
              </div>
              {/* <FadeInSection> */}
                <div className='opening-paragraph'>
                  I'm a full stack software engineer at John Deere. <br/>
                  Some of my key experiences are in implementing event driven systems, managing data platforms, and creating web applications. <br/>
                  Welcome to my website!
                </div>
              {/* </FadeInSection> */}
            </div>
          </span>
        </div>
      </div>
      <div className='img-2'>
        <div className='center'>
          <Resume />
        </div>
      </div>
      <div className='section-1'>
        <div className='default-paragraph'>
          <div>
            I'm creating this page to experiment with interesting React tools. I'll also be including a variety of relevant professional info for myself and resources I've found useful on my journey as a software engineer. So expect updates over the coming months.
          </div>
        </div>
      </div>
      <div className='img-3'>
        <ContactCard />
      </div>
      {/* <div className='img-4'>
      </div> */}
      <div className='built-section'>
        <div>
          <div className='built'>
            Built with 
          </div>
          <div>
            <span>
              <img src={logo} className="App-logo" alt="React logo" href="https://reactjs.org" />
              <img src={nodeLogo} className="default-logo" alt="Node logo" href="https://nodejs.org/en/docs/guides/getting-started-guide/"/>
              <img src={gitLogo} className="default-logo" alt="Github logo" href="https://github.com/cjlludwig"/>
              <img src={dockerLogo} className="default-logo" alt="Docker logo" href="https://www.docker.com/get-started/"/>
              <img src={vsCodeLogo} className="default-logo" alt="Vs Code logo" href="https://code.visualstudio.com/"/>
            </span>
          </div>
          <img style={{padding:"10px"}} src={personalLogo} className="personal-logo" alt="Personal logo" />
        </div>
      </div>
    </div>
  );
}

export default App;
