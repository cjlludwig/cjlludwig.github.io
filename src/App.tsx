import logo from './static/images/logo.svg';
import nodeLogo from './static/images/Node.js_logo.svg';
import gitLogo from './static/images/GitHub_Logo.png';
import dockerLogo from './static/images/docker.svg';
import vsCodeLogo from './static/images/vscode.svg';
import personalLogo from './static/images/logo_transparent512.png';
import awsPic from './static/images/reinvent.jpg';
import googleAnalyticsLogo from './static/images/Logo_Google_Analytics.png';

import ContactCard from './components/ContactCard';
import Resume from './components/Resume';
import Welcome from './components/Welcome';
import Header from './components/Header';

import './static/styles/pages/App.css';
import SpotifyTrends from './components/SpotifyTrends';

function App() {
  return (
    <div className="App">
      <div id="home" className='img-1'>
        <Welcome />
      </div>
      <Header />
      <div id="aboutMe" className="anchor" />
      <div className='section-1'>
        <div className='opening'>
          <span className='opening-span'>
            <div className='img-holder'>
              <img src={awsPic} className="aws-pic" alt="AWS Pic" />
            </div>
            <div className='opening-paragraph-wrapper'>
              <div>
                <b>Connor Ludwig - Staff Software Engineer</b>
              </div>
              <div className='opening-paragraph'>
                I'm a versatile and results-driven Full-stack Software Engineer with over 8 years of experience, currently leading technical initiatives at Built Technologies. My expertise spans:
                <ul>
                  <li>Event-driven architectures and high-throughput systems</li>
                  <li>Cloud-native applications and distributed systems</li>
                  <li>Data platforms and real-time analytics</li>
                  <li>Full-stack development (React, Node.js, Python, Java)</li>
                  <li>Technical leadership and mentorship</li>
                </ul>
                Notable achievements:
                <ul>
                  <li>Led eventing data consistency project unlocking {'>'}5M in ARR</li>
                  <li>Released novel Saas license management system managing {'>'}100M in ARR</li>
                  <li>Mentored many junior engineers and coached them through promotion processes</li>
                </ul>
                I thrive in collaborative environments where I can leverage my technical expertise to mentor team members and drive innovation. My approach combines technical excellence with a focus on delivering business value through scalable, maintainable solutions.
              </div>
            </div>
          </span>
        </div>
      </div>
      <div id="resume" className="anchor" />
      <div className='img-2'>
        <div className='center'>
          <Resume />
        </div>
      </div>
      <div id="listening_to" className="anchor" />
      <div className='section-spotify-trends'>
        <SpotifyTrends />
      </div>
      <div className='section-1'>
        <div className='default-paragraph' style={{paddingTop: '0', paddingBottom: '0'}}>
          <div style={{textAlign: 'center', fontSize: '1.2rem', fontWeight: 500}}>
              <span className='tech-label' style={{display: 'block', marginBottom: '1rem', fontSize: '1.4rem'}}>Certifications</span>
              <span className='certifications'>
                AWS Developer Associate • AWS Solutions Architect Associate • Python for Data Science • Blue River Machine Learning & AI
              </span>
          </div>
        </div>
      </div>
      <div id="contactCard" className="anchor" />
      <div className='img-card'>
        <ContactCard />
      </div>
      <div id="projects" className="anchor" />
      <div className='img-catalog'>
        <div className='projects-section'>
          <h2 className='section-title'>Featured Projects</h2>
          <div className='projects-grid'>
            <div className='project-card'>
              <div className='project-header'>
                <h3>Real-time Financial Processing</h3>
                <span className='project-date'>2024</span>
              </div>
              <div className='project-content'>
                <p>Event-driven solution processing complex financial calculations with focus on accuracy and low-latency. Core project with 1-month turnaround to meet tight product release timeline.</p>
                <div className='tech-stack'>
                  <a href="https://aws.amazon.com" target="_blank" rel="noopener noreferrer">
                    <span>AWS</span>
                  </a>
                  <a href="https://flink.apache.org" target="_blank" rel="noopener noreferrer">
                    <span>PyFlink</span>
                  </a>
                  <a href="https://kafka.apache.org" target="_blank" rel="noopener noreferrer">
                    <span>Kafka</span>
                  </a>
                  <a href="https://aws.amazon.com/lambda" target="_blank" rel="noopener noreferrer">
                    <span>Lambda</span>
                  </a>
                </div>
              </div>
            </div>
            <div className='project-card'>
              <div className='project-header'>
                <h3>Licensing API Platform</h3>
                <span className='project-date'>2021-2023</span>
              </div>
              <div className='project-content'>
                <p>Implemented novel licensing API platform enabling seamless ordering of embedded technologies. Included IoT license delivery, elevated data collection, and expanded product marketing options.</p>
                <div className='tech-stack'>
                  <a href="https://aws.amazon.com" target="_blank" rel="noopener noreferrer">
                    <span>AWS</span>
                  </a>
                  <a href="https://nodejs.org" target="_blank" rel="noopener noreferrer">
                    <span>Node.js</span>
                  </a>
                  <a href="https://developers.google.com/protocol-buffers" target="_blank" rel="noopener noreferrer">
                    <span>Protobuf</span>
                  </a>
                  <a href="https://aws.amazon.com/iot" target="_blank" rel="noopener noreferrer">
                    <span>IoT</span>
                  </a>
                </div>
              </div>
            </div>
            <div className='project-card'>
              <div className='project-header'>
                <h3>Enterprise Data Catalog</h3>
                <span className='project-date'>2019-2021</span>
              </div>
              <div className='project-content'>
                <p>Created modern web application to simplify data discovery, governance, and management for enterprise data lake. App housed 1,000s of datasets with API integrations for data management.</p>
                <div className='tech-stack'>
                  <a href="https://aws.amazon.com" target="_blank" rel="noopener noreferrer">
                    <span>AWS</span>
                  </a>
                  <a href="https://reactjs.org" target="_blank" rel="noopener noreferrer">
                    <span>React</span>
                  </a>
                  <a href="https://nextjs.org" target="_blank" rel="noopener noreferrer">
                    <span>Next.js</span>
                  </a>
                  <a href="https://spark.apache.org" target="_blank" rel="noopener noreferrer">
                    <span>Spark</span>
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div className='built-section'>
        <div>
          <div className='built'>
            Built with 
          </div>
          <div>
            <span>
              <a href="https://reactjs.org">
                <img src={logo} className="App-logo" alt="React logo" />
              </a>
              <a href="https://nodejs.org/en/docs/guides/getting-started-guide/">
                <img src={nodeLogo} className="default-logo" alt="Node logo" />
              </a>
              <a href="https://github.com/cjlludwig">
                <img src={gitLogo} className="default-logo" alt="Github logo" />
              </a>
              <a href="https://www.docker.com/get-started/">
                <img src={dockerLogo} className="default-logo" alt="Docker logo" />
              </a>
              <a href="https://analytics.google.com/analytics/web/">
                <img src={googleAnalyticsLogo} className="default-logo" alt="Google Analytics logo" />
              </a>
              <a href="https://code.visualstudio.com/">
                <img src={vsCodeLogo} className="default-logo" alt="Vs Code logo" />
              </a>
              <a href="https://www.flaticon.com/free-icons/http">
                <img 
                  src="https://media.flaticon.com/dist/min/img/logo/flaticon_negative.svg" 
                  className="default-logo"  
                  alt="Flaticon logo"
                />
              </a>
            </span>
          </div>
          <img style={{padding:"10px"}} src={personalLogo} className="personal-logo" alt="Personal logo" />
          <div className="mobile-spacer"/>
        </div>
      </div>
    </div>
  );
}

export default App;
