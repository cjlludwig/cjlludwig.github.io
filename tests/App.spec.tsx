import { render, screen } from '@testing-library/react';
import renderer from "react-test-renderer";
import App from '../src/App';

import ContactCard from "../src/components/ContactCard"
import Resume from "../src/components/Resume"
import Welcome from "../src/components/Welcome"
import Header from "../src/components/Header"

jest.mock("../src/components/ContactCard");
jest.mock("../src/components/LinkCatalog");
jest.mock("../src/components/Resume");
jest.mock("../src/components/Welcome");
jest.mock("../src/components/Header");

describe("<App />", () => {
  test("renders correctly", () => {
    const tree = renderer
      .create(<App />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
