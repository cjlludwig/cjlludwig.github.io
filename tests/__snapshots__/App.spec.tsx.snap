// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
<div
  className="App"
>
  <div
    className="img-1"
    id="home"
  />
  <div
    className="anchor"
    id="aboutMe"
  />
  <div
    className="section-1"
  >
    <div
      className="opening"
    >
      <span
        className="opening-span"
      >
        <div
          className="img-holder"
        >
          <img
            alt="Wedding Pic"
            className="opening-self-pic"
            src=""
          />
        </div>
        <div
          className="opening-paragraph-wrapper"
        >
          <div>
            <b>
              Connor Ludwig - Staff Software Engineer
            </b>
          </div>
          <div
            className="opening-paragraph"
          >
            I'm a versatile and results-driven Full-stack Software Engineer with over 8 years of experience, currently leading technical initiatives at Built Technologies. My expertise spans:
            <ul>
              <li>
                Event-driven architectures and high-throughput systems
              </li>
              <li>
                Cloud-native applications and distributed systems
              </li>
              <li>
                Data platforms and real-time analytics
              </li>
              <li>
                Full-stack development (React, Node.js, Python, Java)
              </li>
              <li>
                Technical leadership and mentorship
              </li>
            </ul>
            Notable achievements:
            <ul>
              <li>
                Led eventing data consistency project unlocking 
                &gt;
                5M in ARR
              </li>
              <li>
                Released novel Saas license management system managing 
                &gt;
                100M in ARR
              </li>
              <li>
                Mentored many junior engineers and coached them through promotion processes
              </li>
            </ul>
            I thrive in collaborative environments where I can leverage my technical expertise to mentor team members and drive innovation. My approach combines technical excellence with a focus on delivering business value through scalable, maintainable solutions.
          </div>
        </div>
      </span>
    </div>
  </div>
  <div
    className="anchor"
    id="resume"
  />
  <div
    className="img-2"
  >
    <div
      className="center"
    />
  </div>
  <div
    className="anchor"
    id="listening_to"
  />
  <div
    className="section-spotify-trends"
  >
    <div
      className="spotify-trends-container"
    >
      <div
        className="react-multi-carousel-list carousel-container "
        dir="ltr"
      >
        <ul
          className="react-multi-carousel-track "
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={
            {
              "overflow": "unset",
              "transform": "translate3d(0px,0,0)",
              "transition": "none",
            }
          }
        />
      </div>
    </div>
  </div>
  <div
    className="section-1"
  >
    <div
      className="default-paragraph"
      style={
        {
          "paddingBottom": "0",
          "paddingTop": "0",
        }
      }
    >
      <div
        style={
          {
            "fontSize": "1.2rem",
            "fontWeight": 500,
            "textAlign": "center",
          }
        }
      >
        <span
          className="tech-label"
          style={
            {
              "display": "block",
              "fontSize": "1.4rem",
              "marginBottom": "1rem",
            }
          }
        >
          Certifications
        </span>
        <span
          className="certifications"
        >
          AWS Developer Associate • AWS Solutions Architect Associate • Python for Data Science • Blue River Machine Learning & AI
        </span>
      </div>
    </div>
  </div>
  <div
    className="anchor"
    id="contactCard"
  />
  <div
    className="img-card"
  />
  <div
    className="anchor"
    id="projects"
  />
  <div
    className="img-catalog"
  >
    <div
      className="projects-section"
    >
      <h2
        className="section-title"
      >
        Featured Projects
      </h2>
      <div
        className="projects-grid"
      >
        <div
          className="project-card"
        >
          <div
            className="project-header"
          >
            <h3>
              Real-time Financial Processing
            </h3>
            <span
              className="project-date"
            >
              2024
            </span>
          </div>
          <div
            className="project-content"
          >
            <p>
              Event-driven solution processing complex financial calculations with focus on accuracy and low-latency. Core project with 1-month turnaround to meet tight product release timeline.
            </p>
            <div
              className="tech-stack"
            >
              <a
                href="https://aws.amazon.com"
                rel="noopener noreferrer"
                target="_blank"
              >
                <span>
                  AWS
                </span>
              </a>
              <a
                href="https://flink.apache.org"
                rel="noopener noreferrer"
                target="_blank"
              >
                <span>
                  PyFlink
                </span>
              </a>
              <a
                href="https://kafka.apache.org"
                rel="noopener noreferrer"
                target="_blank"
              >
                <span>
                  Kafka
                </span>
              </a>
              <a
                href="https://aws.amazon.com/lambda"
                rel="noopener noreferrer"
                target="_blank"
              >
                <span>
                  Lambda
                </span>
              </a>
            </div>
          </div>
        </div>
        <div
          className="project-card"
        >
          <div
            className="project-header"
          >
            <h3>
              Licensing API Platform
            </h3>
            <span
              className="project-date"
            >
              2021-2023
            </span>
          </div>
          <div
            className="project-content"
          >
            <p>
              Implemented novel licensing API platform enabling seamless ordering of embedded technologies. Included IoT license delivery, elevated data collection, and expanded product marketing options.
            </p>
            <div
              className="tech-stack"
            >
              <a
                href="https://aws.amazon.com"
                rel="noopener noreferrer"
                target="_blank"
              >
                <span>
                  AWS
                </span>
              </a>
              <a
                href="https://nodejs.org"
                rel="noopener noreferrer"
                target="_blank"
              >
                <span>
                  Node.js
                </span>
              </a>
              <a
                href="https://developers.google.com/protocol-buffers"
                rel="noopener noreferrer"
                target="_blank"
              >
                <span>
                  Protobuf
                </span>
              </a>
              <a
                href="https://aws.amazon.com/iot"
                rel="noopener noreferrer"
                target="_blank"
              >
                <span>
                  IoT
                </span>
              </a>
            </div>
          </div>
        </div>
        <div
          className="project-card"
        >
          <div
            className="project-header"
          >
            <h3>
              Enterprise Data Catalog
            </h3>
            <span
              className="project-date"
            >
              2019-2021
            </span>
          </div>
          <div
            className="project-content"
          >
            <p>
              Created modern web application to simplify data discovery, governance, and management for enterprise data lake. App housed 1,000s of datasets with API integrations for data management.
            </p>
            <div
              className="tech-stack"
            >
              <a
                href="https://aws.amazon.com"
                rel="noopener noreferrer"
                target="_blank"
              >
                <span>
                  AWS
                </span>
              </a>
              <a
                href="https://reactjs.org"
                rel="noopener noreferrer"
                target="_blank"
              >
                <span>
                  React
                </span>
              </a>
              <a
                href="https://nextjs.org"
                rel="noopener noreferrer"
                target="_blank"
              >
                <span>
                  Next.js
                </span>
              </a>
              <a
                href="https://spark.apache.org"
                rel="noopener noreferrer"
                target="_blank"
              >
                <span>
                  Spark
                </span>
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="built-section"
  >
    <div>
      <div
        className="built"
      >
        Built with
      </div>
      <div>
        <span>
          <a
            href="https://reactjs.org"
          >
            <img
              alt="React logo"
              className="App-logo"
              src=""
            />
          </a>
          <a
            href="https://nodejs.org/en/docs/guides/getting-started-guide/"
          >
            <img
              alt="Node logo"
              className="default-logo"
              src=""
            />
          </a>
          <a
            href="https://github.com/cjlludwig"
          >
            <img
              alt="Github logo"
              className="default-logo"
              src=""
            />
          </a>
          <a
            href="https://www.docker.com/get-started/"
          >
            <img
              alt="Docker logo"
              className="default-logo"
              src=""
            />
          </a>
          <a
            href="https://analytics.google.com/analytics/web/"
          >
            <img
              alt="Google Analytics logo"
              className="default-logo"
              src=""
            />
          </a>
          <a
            href="https://code.visualstudio.com/"
          >
            <img
              alt="Vs Code logo"
              className="default-logo"
              src=""
            />
          </a>
          <a
            href="https://www.flaticon.com/free-icons/http"
          >
            <img
              alt="Flaticon logo"
              className="default-logo"
              src="https://media.flaticon.com/dist/min/img/logo/flaticon_negative.svg"
            />
          </a>
        </span>
      </div>
      <img
        alt="Personal logo"
        className="personal-logo"
        src=""
        style={
          {
            "padding": "10px",
          }
        }
      />
      <div
        className="mobile-spacer"
      />
    </div>
  </div>
</div>
`;
